### DISCUSSION TITLE
Why does JWK fromKeyLike remove some params (Browser)

### DISCUSSION THREAD

**Discussion by kiehnek** [panva/jose#136] [Q&A] [answered]
Why is the function `[fromKeyLike](https://github.com/panva/jose/blob/v3.5.0/src/runtime/browser/key_to_jwk.ts#L11)` pulling the parameters ext, key_ops, alg & use out of the exported JWK? Especially `alg` would be useful, as it's needed to parse the JWK later.

I don't have a problem with this behavior, but it seems very confusing to me. Is there any downside on having an `alg` param in the JWK? For now i'm partially setting the `alg` header manually or just using `crypto.subtle.exportKey`

**Comment by panva** [comment, marked as answer]
I'm doing this so that the exported object structure is consistent across runtimes.

**Reply by kiehnek** [reply]
That makes sense, thanks for the quick answer .
